src/fondant/pipeline/lightweight_component.py

```diff
@@ -1,23 +1,64 @@
 import inspect
 import itertools
+import logging
+import sys
 import textwrap
 import typing as t
 from dataclasses import asdict, dataclass
 from functools import wraps
+from importlib.metadata import version
 
 from fondant.component import BaseComponent, Component
 
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+MIN_PYTHON_VERSION = (3, 8)
+MAX_PYTHON_VERSION = (3, 11)
+
 
 @dataclass
 class Image:
-    base_image: str = "fondant:latest"
+    base_image: str
     extra_requires: t.Optional[t.List[str]] = None
     script: t.Optional[str] = None
 
     def __post_init__(self):
         if self.base_image is None:
-            # TODO: link to Fondant version
-            self.base_image = "fondant:latest"
+            self.base_image = self.resolve_fndnt_base_image()
+
+        # log info when custom image without Fondant is defined
+        elif self.extra_requires and not any(
+            dependency.startswith("fondant") for dependency in self.extra_requires
+        ):
+            msg = (
+                "You are not using a Fondant default base image, and Fondant is not part of"
+                "your extra requirements. Please make sure that you have installed fondant "
+                "inside your container. Alternatively, you can should add Fondant to "
+                "the extra requirements. \n"
+                "E.g. \n"
+                '@lightweight_component(..., extra_requires=["fondant"])'
+            )
+
+            logger.info(msg)
+
+    @staticmethod
+    def resolve_fndnt_base_image(use_ecr_registry=False):
```

Comment:
This doesn't seem to be called with `use_ecr_registry=True` anywhere.

Reply:
It isn't called. Thought we need it for the case we need it for the sagemaker runner. But I guess its fine to remove it.